text/trace/0170-sampling-probability.md

```diff
@@ -580,92 +586,268 @@ The OTEP 168 proposal for propagation limits head sampling
+bits of information, with 63 ordinary values and a special zero value.
+When `tracestate` is empty, the `ot` subkey cannnot be found, or the
+`p` value cannot be determined, the head sampling probability is
+considered unknown.
 
 | Value | Head Adjusted Count |
 | ----- | ---------------- |
-| 0 | _Unknown_ |
-| 1 | 1 |
-| 2 | 2 |
-| 3 | 4 |
-| 4 | 8 |
-| 5 | 16 |
-| 6 | 32 |
+| 0 | 1 |
+| 1 | 2 |
+| 2 | 4 |
+| 3 | 8 |
+| 4 | 16 |
 | ... | ... |
-| X | 2^(X-1) |
+| X | 2**X |
 | ... | ... |
-| 62 | 2^61 |
+| 62 | 2**62 |
 | 63 | 0 |
 
 Combined with the proposal for propagating head sampling probability
 in OTEP 168, the result is that Sampling can be enabled in an
-up-to-date system and all Spans, roots and children alike, will have a
-non-zero values in the `log_head_adjusted_count` field.  Consumers of a
-stream of Span data with non-zero values in the `log_head_adjusted_count`
-field can approximately and accurately count Spans using adjusted
-counts.
+up-to-date system and all Spans, roots and children alike, will have
+known adjusted count.  Consumers of a stream of Span data with the
+OTEP 168 `tracestate` value can approximately and accurately count
+Spans by their adjusted count.
+
+### Span data model changes
+
+TODO: See https://github.com/open-telemetry/opentelemetry-specification/issues/1932
+
+Addition to the Span data model:
+
+```
+### Definitions Used in this Document
+
+#### Sampler
+
+A Sampler provides configurable logic, used by the SDK, for selecting
+which Spans are "recorded" and/or "sampled" in a tracing client
+library.  To "record" a span means to build a representation of it in
+the client's memory, which makes it eligible for being exported.  To
+"sample" a span implies setting the W3C `sampled` flag and recording
+the span for export.
+
+OpenTelemetry supports spans that are "recorded" and not "sampled"
+for "live" observability of spans (e.g., z-pages).
+
+#### Parent-based sampling
+
+A Sampler that makes its decision to sample based on the W3C `sampled`
+flag is said to use parent-based sampling.
+
+#### Head sampling
+
+In a tracing context, Head sampling refers to the initial decision to
+sample a span or a trace, which determines the W3C `sampled` flag of
+the child context.  The OpenTelemetry tracing data model currently
+supports only head sampling.
+
+#### Probability sampler
+
+A probability Sampler is a Sampler that knows immediately, for each
+of its decisions, the probability that the span had of being selected.
+
+Sampling probability is defined as a number less than or equal to 1
+and greater than 0 (i.e., `0 < probability <= 1`).  The case of 0
+probability is treated as a special, non-probabilistic case.
+
+#### Consistent probability sampler
+
+A consistent probability sampler is a Sampler that for a configured
+sampling probability will make the same decision as another sampler
+with the same or greater probability.  In OpenTelemetry, consistent
```

Comment:
This is not precise enough. A decision can be "to sample" or "not to sample", it should be made clear that only the former kind is considered in this sentence.

Reply:
Could you help me understand what is missing? Since the decision to sample or not to sample is binary, isn't the decision equivalent either way? 

There is a general quesatsion in OpenTelemetry right now over whether the Sampler API should be viewed as an exclusive mechanism for selecting which spans are exported. The rules defined here are trying to decouple two things:
(1) how we select which spans are exported, and at what level of detail (I'm referring to as "View configuration")
(2) how we (estimate) the count of spans that are exported after sampling has been applied.

Reply:
Let me try to express my remark differently. A consistent probability sampler as defined by your sentence will always be equivalent (with respect to all its sampling decisions) to a probability sampler configured with P=1. In other words, its configured probability must have no effect on sampling decisions - this certainly was not the intention.

Reply:
How does this sound:
```
A consistent probability sampler is a Sampler that supports independent
sampling decisions at each span in a trace while maintaining that 
traces will be complete with probability equal to the minimum sampling 
probability across the trace.  Consistent probability sampling requires
for a span with lesser sampling probability `probability<sub>1</sub>` 
and a span with greater sampling probability `probability<sub>2</sub>` 
if the lesser-probability span is sampled the greater-probability span 
MUST also be sampled.
```

Reply:
Yes, this is much better, but the wording implies that "probability" is tied to "spans", which may be confusing. May I suggest the second sentence to be

Consistent probability sampling requires that for each given span, if a Sampler with
lesser sampling probability `probability<sub>1</sub>` selects it for sampling,
then the span will be also always selected when the Sampler is configured with 
greater sampling probability `probability<sub>2</sub>`.

Reply:
I want to be sure this applies to any span in a trace, so I've revised your suggestion further:
```
Consistent probability sampling requires that 
for any span in a given trace, if a Sampler with lesser sampling probability 
selects the span for sampling, then the span would also be selected by a
Sampler configured with greater sampling probability.
```

Reply:
Now it looks even better, thanks.